### PR TITLE
12003 - OldDeckName now supported as a piece property

### LINKED ISSUE #11997: OldDeckName property (request)
OldDeckName would be set to the current value of DeckName whenever a piece would set other "Old" location properties such as OldMap etc.

Previously raised on the [forum](https://forum.vassalengine.org/t/olddeckname-property/75359).

### PR DESCRIPTION
**Pull request by Cattlesquat**
Fixes #11997

### CODE REVIEW

**Comment by Cattlesquat** [comment]
@riverwanderer OldDeckName ready for test...

**Comment by riverwanderer** [comment]
Tested successfully with the C&C Ancient Module. OldDeckName is set on movement, including being nulled out if the movement was from a non-deck location.

File: vassal-doc/src/main/readme-referencemanual/ReferenceManual/BasicPiece.adoc [hunk 1]
```diff
@@ -68,6 +68,7 @@ The following <<Properties.adoc#top,Properties>> are defined after a piece is mo
 * _OldZone_ contains the name of the previous <<ZonedGrid.adoc#top,Zone>>
 * _OldX_ contains the previous map X coordinate
 * _OldY_ contains the previous map Y coordinate
+* _OldDeckName_ contains the name of any deck the piece was in before its last move (will be an empty string piece wasn't in a deck)
```
Comment (jonathanrwatts) [inline vassal-doc/src/main/readme-referencemanual/ReferenceManual/BasicPiece.adoc:71]: ...empty string *if* piece....

File: vassal-doc/src/main/readme-referencemanual/ReferenceManual/Properties.adoc [hunk 2]
```diff
@@ -79,6 +79,7 @@ the level in a <<Layer.adoc#Properties,Layer)>> you can use _<_, _\<=_, _>_, and
 |*Obscured* |<<Mask.adoc#top,Mask>> |"true" if this piece is masked from the current player.
 |*ObscuredToOthers* |<<Mask.adoc#top,Mask>> |"true" if this piece is masked from other players.
 |*OldBoard* |<<BasicPiece#top,Basic Piece>> |Board name prior to most recent movement.
+|*OldDeckName* |<<BasicPiece#top,Basic Piece>> |Name of any <<Deck.adoc#top,Deck>> the piece was in before its last move (will be an empty string piece wasn't in a deck)
```
Comment (jonathanrwatts) [inline vassal-doc/src/main/readme-referencemanual/ReferenceManual/Properties.adoc:82]: ...empty string *if* piece....

Comment (Cattlesquat) [reply vassal-doc/src/main/readme-referencemanual/ReferenceManual/Properties.adoc:82]: Thanks!

**Pull request merged**